### ISSUE TITLE
Docs: remove old content from 'import_export/'

### ISSUE THREAD

**Issue by ramirezfranciscof** [aiidateam/aiida-core#4450]
This is a structure that was left over from the old documentation and needs to be removed, but first we need to check if all the content has been re-assigned in the new documentation. The files considered are:

- [ ] [main.rst](https://aiida-core.readthedocs.io/en/v1.1.0/import_export/main.html)
- [ ] [dbimporters/cod.rst](https://aiida-core.readthedocs.io/en/v1.1.0/import_export/external_dbs.html)
- [ ] [dbimporters/icsd.rst](https://aiida-core.readthedocs.io/en/v1.1.0/import_export/external_dbs.html)
- [ ] [external_dbs.rst](https://aiida-core.readthedocs.io/en/v1.1.0/import_export/external_dbs.html)

### Proposition:

The first "main" page (about the export format) we already decided to move to "Internals: Data storage (Export archive)" (see [here](https://github.com/aiidateam/aiida-core/issues/4037)). The other three I think would be better suited for a "howto/import_from_online_dbs" section.

**Comment by mbercx** [comment]
> The first "main" page (about the export format) we already decided to move to "Internals: Data storage (Export archive)" (see [here](https://github.com/aiidateam/aiida-core/issues/4037)). 

Perhaps how to export data also belongs in the "How to work with data" section? There is still a section marked as TODO on "How to share data" (see also #3998).

> The other three I think would be better suited for a "howto/import_from_online_dbs" section.

Should this also be a part of "How to work with data" -> ["Importing data"](https://aiida.readthedocs.io/projects/aiida-core/en/latest/howto/data.html#importing-data), under a subsection "Importing from external databases"?

**Comment by sphuber** [comment]
Section "Export File Format" should go to the "Internals: Data storage" section, for which there already is issue #4037
The other sections "Export", "Import", "Export Archive Migration" and "Inspect Export Archives" should be moved together to the "How to - share data" section. Note that this section is targeted at end users that just want to use it, so any technical background information that may be present in the existing sections should be removed or reworded.

**Comment by sphuber** [comment]
The page on external importers should probably not go directly in "Importing Data" because this is a lot of information that is quite specific and it will break the flow of the entire page, which is meant to be more generic. However, the content is still important so we propose to create a new page that is dedicated to docs on dbimporters that ship with `aiida-core`, which is linked to from the "Importing data" section, with maybe a short preface of explaining how the `DbImporter` concept works and can be implemented for any external database

**Comment by giovannipizzi** [comment]
One important additional thing for the [ICSD importer page](https://aiida-core.readthedocs.io/en/v1.1.0/import_export/dbimporters/icsd.html): let's add a note on the top that the implementation works only if you buy the ICSD CD/DVD and setup the MySQL DB on some machine - access through the online web interface at https://icsd.products.fiz-karlsruhe.de (after clicking at the login button) is not supported

**Comment by ramirezfranciscof** [comment]
> However, the content is still important so we propose to create a new page that is dedicated to docs on dbimporters that ship with aiida-core, which is linked to from the "Importing data" section, with maybe a short preface of explaining how the DbImporter concept works and can be implemented for any external database

So, if I understand correctly this page would have this linked pointed at it but would otherwise be..."loose"? (as in, not indexed under any of the current sections)

**Comment by CasperWA** [comment]
> Section "Export File Format" should go to the "Internals: Data storage" section, for which there already is issue #4037

How about creating a section in `Reference`, e.g., `AiiDA archive format`?

**Comment by ramirezfranciscof** [comment]
The two linked PRS should be taking care of this:

#4467: section "Export File Format" in `main.rst`
#4458: other sections in `main.rst` and all `external_dbs.rst`, `dbimporters/cod.rst` and `dbimporters/icsd.rst`

**Comment by ltalirz** [comment]
Just to mention that #4458 will take care of the second item on the to-do list but doesn't unearth the instructions for the ICSD/COD importers.

As far as I can see, there are actually many more importers now, but except for the cod importer they are rather poorly tested, so I'm not sure they actually work:
https://codecov.io/gh/aiidateam/aiida-core/tree/5e0de12cd21d18da1e57c87f8f5e0130edda508d/aiida/tools/dbimporters/plugins

After thinking a bit about where to place the corresponding documentation, I propose to
 * leave it dormant for the moment until we...
 * move the dbimporters and dbexporters to a separate plugin.

It seems to me that the importers/exporters are a good candidate for moving to a plugin - they are clearly adding functionality on top of AiiDA and they aren't deeply fused to the API, so one can probably even do this in a backward-compatible way (adding the plugin package as a dependency for the moment).

It doesn't have to be done immediately, but I propose to decide whether we are going to do this and, if yes, to postpone any update of the corresponding documentation until that point.

**Comment by sphuber** [comment]
I think moving the importers to a plugin makes sense. Especially the COD importer would be a great fit for `aiida-codtools` as it has additional functionality that works directly with it, such as CLI to import from it. The ICSD and MPDS one could then also be migrated there. I am not familiar with the other ones.

**Comment by CasperWA** [comment]
Indeed, they seem perfect candidates for a separate plugin, however, they are closely related to the materials science aspect of AiiDA, and as such I think we should be careful to move it out when other materials science-related parts still remain in AiiDA Core (`StructureData`, `BandsData`, ...) Unless the plug-in(s) will always be included in the `atomic_tools` extra?

**Comment by sphuber** [comment]
I think it was implied that they would be moved out together with the other domain specific plugins, which would be for v2.0

**Comment by ltalirz** [comment]
The dbimporters/dbexporters are a bit different from the `Data` classes in that here we don't have the (I feel very serious) issue of losing the ability to do data migrations for any data type moved outside aiida-core.

In that sense, one could decide to use the importers/exporters as an earlier test case of such a migration (also because they are used much less than the built-in data classes), but at the same time one may also prefer to do everything in one go.
